### DISCUSSION TITLE
No such file or directory (os error 2) problems in Production builds

### DISCUSSION THREAD

**Discussion by Blatman** [tauri-apps/tauri#5443] [Q&A] [answered]
getting errors when paths include ../ -  ERR- "No such file or directory (os error 2)"

        resolvePath gives same errors as resourceDir.  
		
        const resourceDir = await path.resourceDir()  // this gives ... winman.app/Contents/MacOS/../Resources/lang/de.json" ????
		var correctedResourceDir = resourceDir.replace(/MacOS\/\.\.\/Resources\//, 'Resources/');  // now is  ... winman.app/Contents/Resources/lang/de.json"
		var resourcePath = correctedResourceDir;
		
		Corrected path will work in both Dev and Prod builds whereas path with ../ included will only work in Dev builds.
		
		
		Using `await fs.readTextFile('lang/de.json', {dir: fs.BaseDirectory.Resource})` works OK.
		
		
		Also get these errors in Prod builds when using child execute processes
		
		eg:
		
		var output = await new cmd('list-dir').execute();		// ls, chmod work OK in Dev and Prod
        tauriinfo = output.stdout;
		
        var output = await new cmd('npm-version', ['version']).execute();  //  npm fails in Prod, OK in Dev
        tauriinfo = output.stdout;		 
		
		
		"scope": [
		  {
          "name": "list-dir",
          "cmd": "ls",
		  "sidecar": false,
          "args": false
          },
		  {
          "name": "npm-version",
          "cmd": "npm",
		  "sidecar": false,
          "args": true
          },

Came across something about fix-path which sounds similar but had no luck with implementing the suggested methods (couldn't actually find the crate that is supposed to exist). Not sure if the fix has been fully integrated into the latest Tauri build and this is another issue.

**Comment by FabianLars** [comment, marked as answer]
That's a bug fixed in https://github.com/tauri-apps/tauri/pull/5218 - should be released relativelyyy soon

**Reply by Blatman** [reply]
Ah excellent  - finding it a bit of a showstopper using the shell child process commands - will try to wait very patiently ...

**Comment by Blatman** [comment]
Excellent - v1.2.0 resolves the path issue.
